src/Symfony/Component/ErrorHandler/Resources/views/traces.html.php

```diff
@@ -1,21 +1,17 @@
 <div class="trace trace-as-html" id="trace-box-<?= $index; ?>">
     <div class="trace-details">
         <div class="trace-head">
-            <span class="sf-toggle" data-toggle-selector="#trace-html-<?= $index; ?>" data-toggle-initial="<?= $expand ? 'display' : ''; ?>">
-                <h3 class="trace-class">
-                    <span class="icon icon-close"><?= $this->include('assets/images/icon-minus-square-o.svg'); ?></span>
-                    <span class="icon icon-open"><?= $this->include('assets/images/icon-plus-square-o.svg'); ?></span>
-
-                    <span class="trace-namespace">
-                        <?= implode('\\', array_slice(explode('\\', $exception['class']), 0, -1)); ?><?= count(explode('\\', $exception['class'])) > 1 ? '\\' : ''; ?>
-                    </span>
-                    <?= ($parts = explode('\\', $exception['class'])) ? end($parts) : ''; ?>
-                </h3>
-
+            <div class="sf-toggle" data-toggle-selector="#trace-html-<?= $index; ?>" data-toggle-initial="<?= $expand ? 'display' : ''; ?>">
+                <span class="icon icon-close"><?= $this->include('assets/images/icon-minus-square-o.svg'); ?></span>
+                <span class="icon icon-open"><?= $this->include('assets/images/icon-plus-square-o.svg'); ?></span>
+                <h4 class="trace-namespace">
+                    <?= implode('\\', array_slice(explode('\\', $exception['class']), 0, -1)); ?><?= count(explode('\\', $exception['class'])) > 1 ? '\\' : ''; ?>
+                </h4>
+                <h3 class="trace-class"><?= ($parts = explode('\\', $exception['class'])) ? end($parts) : ''; ?></h3>
                 <?php if ($exception['message'] && $index > 1) { ?>
-                    <p class="break-long-words trace-message"><?= $this->escape($exception['message']); ?></p>
+                    <h4 class="break-long-words trace-message"><?= $this->escape($exception['message']); ?></h4>
```

Comment:
Should the exception message really become a heading ?

Reply:
Yes, it's in the same header-box/div/gray-box-thing as the FQCN and is definitely not a paragraph

Reply:
From an accessibility point of view, the message is definitely not a heading. The choice between `<h4>` and `<p>` should not be done based on styling but on semantic.

Reply:
> From an accessibility point of view, the message is definitely not a heading.

Sorry for asking. But could you elaborate on this?

Reply:
if you make both the class name and the message be heading, you end up with a content being only headings, and these headings have no content that they are heading.
Btw, you also end up with a single h4-level subsection in each h3-level section, which tends to also indicate that this is not proper subsections of the content (even when forgetting the fact that the content is empty).

See https://hiddedevries.nl/en/blog/2020-09-05-when-there-is-no-content-between-headings talking about this.

Reply:
Thank you for the clarification.

@awoimbee, could you change it back to `p`?

Reply:
Done !